client/tink_keyset_helper.go

```diff
@@ -80,3 +81,43 @@ func convertTinkKeysetHandleToBytes(keysetHandle *keyset.Handle) ([]byte, error)
 	}
 	return bytesBuffer.Bytes(), nil
 }
+
+// addNewTinkKeyset receives a knox version list and a tink key templateFunc, create a new tink keyset contains
+// a single fresh key from the given tink key templateFunc. Most importantly, the ID of this signle fresh key is
+// different from the ID of all existed tink keys in the given knox version list (avoid Tink key ID duplications).
+func addNewTinkKeyset(templateFunc func() *tinkpb.KeyTemplate, knoxVersionList knox.KeyVersionList) ([]byte, error) {
+	existedTinkKeysID := make(map[uint32]struct{})
+	for _, v := range knoxVersionList {
+		tinkKeysetForAVersion, err := readTinkKeysetFromBytes(v.Data)
+		if err != nil {
+			return nil, err
+		}
+		existedTinkKeysID[tinkKeysetForAVersion.PrimaryKeyId] = struct{}{}
+	}
+	var keysetHandle *keyset.Handle
+	var err error
+	for {
+		keysetHandle, err = keyset.NewHandle(templateFunc())
+		if keysetHandle == nil || err != nil {
+			return nil, fmt.Errorf("cannot get tink keyset handle: %v", err)
+		}
+		newTinkKeyID := keysetHandle.KeysetInfo().PrimaryKeyId
+		// Check whether the ID of created tink key to avoid key ID duplication
+		_, ok := existedTinkKeysID[newTinkKeyID]
+		if !ok {
```

Comment:
even though ok is the convention, duplicate is probably a more readable name for this specific case.

Reply:
Thank you so much for your comment! I have revised the code.